lib/matplotlib/colors.py

```diff
@@ -234,6 +227,16 @@ def to_rgba_array(c, alpha=None):
         if np.any((result < 0) | (result > 1)):
             raise ValueError("RGBA values should be within 0-1 range")
         return result
+    # Handle single values.
+    # Note that this occurs *after* handling inputs that are already arrays, as
+    # `to_rgba(c, alpha)` (below) is expensive for such inputs, due to the need
+    # to format the array in the ValueError message(!).
+    if isinstance(c, six.string_types) and c.lower() == "none":
+        return np.zeros((0, 4), float)
```

Comment:
It comes down to whether you want to fast-track array inputs or "none" -- for both inputs, the other path gets rejected after just an instance check.
So the choice to move it down was just to keep the cases handling size-1 returns together.